File: VB/RangedMaskSample/MaskButtonEditMaskFormatInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Data.Mask
Imports System.Globalization

Namespace RangedMaskSample
	Public Class MaskButtonEditMaskFormatInfo
		Inherits DateTimeMaskFormatInfo
		Public Sub New(ByVal mask As String, ByVal dateTimeFormatInfo As DateTimeFormatInfo, ByVal aedit As MaskButtonEdit)
			MyBase.New(mask, dateTimeFormatInfo)
			For i As Integer = 0 To Count - 1
                If (i < Count - 1) AndAlso (TypeOf innerList(i) Is DateTimeMaskFormatElementLiteral) AndAlso (TypeOf innerList(i + 1) Is DateTimeMaskFormatElementLiteral) Then
                    Dim aMaskSymbol As String = (CType(innerList(i), DateTimeMaskFormatElementLiteral)).Literal + (CType(innerList(i + 1), DateTimeMaskFormatElementLiteral)).Literal
                    Dim aElement As MaskElement = MaskElementsRepository.MaskCollection.Find(aMaskSymbol)
                    If aElement IsNot Nothing Then
                        innerList(i) = New MaskButtonEditMaskFormatElement(aMaskSymbol, dateTimeFormatInfo, aedit.Properties.MaskElements.IndexOf(aElement), aedit)
                        innerList.RemoveAt(i + 1)
                    End If
                End If
			Next i
		End Sub
	End Class
End Namespace

File: CS/RangedMaskSample/MaskButtonEditMaskFormatInfo.cs
using System;
using DevExpress.Data.Mask;
using System.Globalization;

namespace RangedMaskSample
{
    public class MaskButtonEditMaskFormatInfo : DateTimeMaskFormatInfo
    {
        public MaskButtonEditMaskFormatInfo(string mask, DateTimeFormatInfo dateTimeFormatInfo, MaskButtonEdit aedit)
            : base(mask, dateTimeFormatInfo)
        {
            for (int i = 0; i < Count; i++)
            {
                if ((innerList[i] is DateTimeMaskFormatElementLiteral) && (i < Count - 1) && (innerList[i + 1] is DateTimeMaskFormatElementLiteral))
                {
                    string aMaskSymbol = ((DateTimeMaskFormatElementLiteral)innerList[i]).Literal +
                        ((DateTimeMaskFormatElementLiteral)innerList[i + 1]).Literal;
                    MaskElement aElement = MaskElementsRepository.MaskCollection.Find(aMaskSymbol);
                    if (aElement != null)
                    {
                        innerList[i] = new MaskButtonEditMaskFormatElement(aMaskSymbol, dateTimeFormatInfo, aedit.Properties.MaskElements.IndexOf(aElement), aedit);
                        innerList.RemoveAt(i + 1);
                    }
                }
            }
        }
    }
}
